File: gunzip/ruby/gunzip_benchmark.rb
#!/usr/bin/env ruby
require "zlib"
start_time = Time.now
uncompressed_data = nil
Zlib::GzipReader.open("../out/nist/2011.json.gz") { | gz |
  uncompressed_data = gz.read
}
end_time = Time.now
puts("Ruby: Decompressing took #{((end_time - start_time)*1000).to_i}ms to #{uncompressed_data.size} bytes")

File: gunzip/python/gunzip_benchmark.py
#!/usr/bin/env python3
import gzip
import time

start_time = time.time()
with gzip.open('../out/nist/2011.json.gz') as f:
    uncompressed_data = f.read();
end_time = time.time()
print("- Python: Decompressing took {} ms to {} bytes".format(int((end_time-start_time)*1000), len(uncompressed_data)))
